operators/src/pro/cache/shared_cache.rs

```diff
@@ -0,0 +1,1190 @@
+                            .raster_caches
+                            .get_mut(&raster_pop_key)
+                            .expect("LRU entry must exist in the cache!");
+                        let query_element = op_cache
+                            .remove_cache_entry(&pop_id)
+                            .expect("LRU entry must exist in the cache!");
+                        self.cache_size.remove_element_bytes(&query_element);
+                    }
+                    TypedCanonicOperatorName::Vector(vector_pop_key) => {
+                        let op_cache = self
+                            .vector_caches
+                            .get_mut(&vector_pop_key)
+                            .expect("LRU entry must exist in the cache!");
+                        let query_element = op_cache
+                            .remove_cache_entry(&pop_id)
+                            .expect("LRU entry must exist in the cache!");
+                        self.cache_size.remove_element_bytes(&query_element);
+                    }
+                };
+                self.cache_size.remove_element_bytes(&pop_id);
+
+                debug!(
+                    "Evicted query {}. Cache size: {}. Cache size used: {}, Cache used percentage: {}.",
+                    pop_id,
+                    self.cache_size.total_byte_size(),
+                    self.cache_size.byte_size_used(),
+                    self.cache_size.size_used_fraction()
+                );
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum TypedCanonicOperatorName {
+    Raster(CanonicOperatorName),
+    Vector(CanonicOperatorName),
+}
+
+impl TypedCanonicOperatorName {
+    pub fn as_raster(&self) -> Option<&CanonicOperatorName> {
+        match self {
+            Self::Raster(name) => Some(name),
+            Self::Vector(_) => None,
+        }
+    }
+
+    pub fn as_vector(&self) -> Option<&CanonicOperatorName> {
+        match self {
+            Self::Raster(_) => None,
+            Self::Vector(name) => Some(name),
+        }
+    }
+}
+
+pub trait CacheEvictUntilFit {
+    fn evict_entries_until_can_fit_bytes(&mut self, bytes: usize);
+}
+
+impl CacheEvictUntilFit for CacheBackend {
+    fn evict_entries_until_can_fit_bytes(&mut self, bytes: usize) {
+        self.evict_until_can_fit_bytes(bytes);
+    }
+}
+
+pub trait CacheView<C, L>: CacheEvictUntilFit {
+    fn operator_caches_mut(
+        &mut self,
+    ) -> &mut HashMap<CanonicOperatorName, OperatorCacheEntry<C, L>>;
+
+    fn create_operator_cache_if_needed(&mut self, key: CanonicOperatorName) {
+        self.operator_caches_mut()
+            .entry(key)
+            .or_insert_with(|| OperatorCacheEntry::new());
+    }
+
+    fn remove_operator_cache(
+        &mut self,
+        key: &CanonicOperatorName,
+    ) -> Option<OperatorCacheEntry<C, L>> {
+        // TODO: maybe remove the size of the OperatorCacheEntry to the cache size?
```

Comment:
do you add the size to the cache size in the first place, though?

Reply:
nope thats also a TODO

Reply:
[2d219f8](https://github.com/geo-engine/geoengine/pull/833/commits/2d219f86e20ee1bbe169d7a7248edf1f20a71927)